scripts/src/javatests/org/oppia/android/scripts/todo/TodoCheckTest.kt

```diff
@@ -0,0 +1,422 @@
+      """.trimIndent()
+    val testJSONFile = tempFolder.newFile("testfiles/open_issues.json")
+    testJSONFile.writeText(testJSONContent)
+    val tempFile1 = tempFolder.newFile("testfiles/TempFile1.kt")
+    val tempFile2 = tempFolder.newFile("testfiles/TempFile2.kt")
+    val testContent1 =
+      """
+      // TODO (152440222): test description 1
+      """.trimIndent()
+    val testContent2 =
+      """
+      # TODO(#1000000): test description 2
+      """.trimIndent()
+    tempFile1.writeText(testContent1)
+    tempFile2.writeText(testContent2)
+    val exemptionFile = File("${tempFolder.root}/$pathToProtoBinary")
+    val exemptions = TodoExemptions.newBuilder().apply {
+      this.addAllTodoExemption(
+        listOf(
+          TodoExemption.newBuilder().apply {
+            this.exemptedFilePath = "TempFile1.kt"
+            this.addAllLineNumber(listOf(1)).build()
+          }.build(),
+          TodoExemption.newBuilder().apply {
+            this.exemptedFilePath = "TempFile2.kt"
+            this.addAllLineNumber(listOf(1)).build()
+          }.build()
+        )
+      )
+    }.build()
+    exemptions.writeTo(exemptionFile.outputStream())
+
+    runScript()
+
+    assertThat(outContent.toString().trim()).isEqualTo(TODO_CHECK_PASSED_OUTPUT_INDICATOR)
+  }
+
+  @Test
+  fun testTodoCheck_allTodosAreValid_redundantExemption_checkShouldFail() {
+    val testJSONContent =
+      """
+      [{"number":1000000},{"number":152440222},{"number":152440223},{"number":11001}]
+      """.trimIndent()
+    val testJSONFile = tempFolder.newFile("testfiles/open_issues.json")
+    testJSONFile.writeText(testJSONContent)
+    val tempFile1 = tempFolder.newFile("testfiles/TempFile1.kt")
+    val tempFile2 = tempFolder.newFile("testfiles/TempFile2.kt")
+    val testContent1 =
+      """
+      // TODO(#152440222): test description 1
+      // TODO(#152440223): test description 1
+      """.trimIndent()
+    val testContent2 =
+      """
+      # TODO(#1000000): test description 2
+      """.trimIndent()
+    tempFile1.writeText(testContent1)
+    tempFile2.writeText(testContent2)
+    val exemptionFile = File("${tempFolder.root}/$pathToProtoBinary")
+    val exemptions = TodoExemptions.newBuilder().apply {
+      this.addAllTodoExemption(
+        listOf(
+          TodoExemption.newBuilder().apply {
+            this.exemptedFilePath = "TempFile1.kt"
+            this.addAllLineNumber(listOf(1, 2)).build()
+          }.build(),
+          TodoExemption.newBuilder().apply {
+            this.exemptedFilePath = "TempFile2.kt"
+            this.addAllLineNumber(listOf(1)).build()
+          }.build()
+        )
+      )
+    }.build()
+    exemptions.writeTo(exemptionFile.outputStream())
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(TODO_SYNTAX_CHECK_FAILED_OUTPUT_INDICATOR)
+    val failureMessage =
+      """
+      Redundant exemptions:
```

Comment:
Maybe update to be clearer on what is redundant here:

```suggestion
      Redundant exemptions (there are no TODOs corresponding to these lines):
```

Reply:
Done.